src/renderers/shared/utils/__tests__/ReactErrorUtils-test.js

```diff
@@ -132,36 +135,56 @@ describe('ReactErrorUtils', () => {
     it(`does not return nested errors (${environment})`, () => {
       const err1 = new Error();
       let err2;
-      const err3 = ReactErrorUtils.invokeGuardedCallback(
+      ReactErrorUtils.invokeGuardedCallback(
         'foo',
         function() {
-          err2 = ReactErrorUtils.invokeGuardedCallback(
+          ReactErrorUtils.invokeGuardedCallback(
             'foo',
             function() {
               throw err1;
             },
             null,
           );
+          err2 = ReactErrorUtils.clearCaughtError();
         },
         null,
       );
+      // Returns null because inner error was already captured
+      expect(ReactErrorUtils.hasCaughtError()).toBe(false);
 
-      expect(err3).toBe(null); // Returns null because inner error was already captured
       expect(err2).toBe(err1);
     });
 
+    if (!__DEV__) {
+      // jsdom doesn't handle this properly, but Chrome and Firefox should. Test
+      // this with a fixture.
```

Comment:
How does this work? We'll try to run tests in prod mode too even in jest so JSDOM needs to handle it. Also, we don't run jest in browsers so how does this get run in a browser?